data/ha/hawk_test_ssh.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/python3
+"""Define SSH related functions to test the HAWK GUI"""
+
+from distutils.version import LooseVersion as Version
+import paramiko, hawk_test_results, warnings
+# Ignore CryptographyDeprecationWarning shown when using paramiko
+try:
+    from cryptography.utils import CryptographyDeprecationWarning
+    warnings.simplefilter('ignore', CryptographyDeprecationWarning)
+except ImportError:
+    pass
+
+class hawkTestSSHError(Exception):
+    """Base class for exceptions in this module."""
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return repr(self.value)
+
+class hawkTestSSH:
+    def __init__(self, hostname, secret):
+        self.ssh = paramiko.SSHClient()
+        self.ssh.load_system_host_keys()
+        self.ssh.set_missing_host_key_policy(paramiko.WarningPolicy)
+        self.ssh.connect(hostname=hostname.lower(), username="root", password=secret)
+
+    def is_ssh(self):
+        clase = str(type(self.ssh)).split(' ')[1][1:26]
+        if clase != 'paramiko.client.SSHClient':
+            raise hawkTestSSHError('SSH object must be of type %s. Got: [%s]' %
+                                   ('paramiko.client.SSHClient', type(s)))
+        return True
+
+    def check_cluster_conf_ssh(self, command, mustmatch):
+        command = str(command)
+        if self.is_ssh():
+            resp = self.ssh.exec_command(command)
+            out = resp[1].read().decode().rstrip('\n')
+            err = resp[2].read().decode().rstrip('\n')
+            print("INFO: ssh command [%s] got output [%s] and error [%s]" % (command, out, err))
+            if err:
+                print("ERROR: got an error over SSH: [%s]" % err)
+                return False
+            if isinstance(mustmatch, str):
+                if mustmatch:
+                    if out.find(mustmatch) >= 0:
+                        return True
+                    return False
+                return out == mustmatch
+            elif isinstance(mustmatch, list):
+                for exp in mustmatch:
+                    if out.find(str(exp)) < 0:
```

Comment:
```suggestion
                    if str(exp) not in out:
```
Pythonic way to test for a substring.

Reply:
Fixing.